src/test/java/com/google/sps/SessionContextTest.java

```diff
@@ -0,0 +1,78 @@
+package com.google.sps;
+
+import static org.mockito.Mockito.when;
+
+import com.google.sps.data.SessionContext;
+import com.google.sps.data.NonPersistentUserRepository;
+import com.google.sps.data.UserRepository;
+import com.google.sps.data.User;
+import java.lang.Exception;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class SessionContextTest {
+
+    private SessionContext sessionContext;
+    
+    @Before
+    public void setup() {
+        sessionContext = Mockito.mock(SessionContext.class);
+    }
+
+    @Test
+    public void getLoggedInUser_ReturnUser() {
```

Comment:
Unless I'm misreading, these tests end up not running/testing the lines in SessionContext since the mocks override the implementations (you should be able to verify this by breaking the logic in SessionContext (but in a compilable way), and seeing that the tests still pass);

In general, you want your mocks/fakes/stubs to override the _components_ (usually member variables) your module depends on, so you can still run/test the logic in your module, but also avoid running all the dependencies.

To test your current design, this would entail:
- Making the sessionContext (L23) into an actual instance of SessionContext (so, for instance, calls to 'getLoggedInUsers' as on L31 run on the instance, rather than just querying a mock object)
- Initializing your real instance of SessionContext (first item) with a mock UserRepository, saved as a private variable in the setup for use in the tests
- Injecting a mock userService (you can either take one in a new constructor overload, or write a method like testOnlySetUserService that injects the mock specifically for the test). You'll also need this in the tests.
- Setting your when ... thenReturn expectations on top of those mocks, rather than on 'sessionContext' (which now is a concrete instance).

Reply:
I think it is fixed now! Thanks for the comment, it was really easy to follow :)